onboard/nucleo_bridge/Frontend.h

```diff
@@ -0,0 +1,197 @@
+	    (*controllers)["RA_4"]->open_loop(msg->throttle[4]);
+	    (*controllers)["RA_5"]->open_loop(msg->throttle[5]);
+	    ra_pos_data();
+	}
+
+	void sa_open_loop_cmd(LCM_INPUT, const SAOpenLoopCmd *msg){		
+	    (*controllers)["SA_0"]->open_loop(msg->throttle[0]);
+	    (*controllers)["SA_1"]->open_loop(msg->throttle[1]);
+	    (*controllers)["SA_2"]->open_loop(msg->throttle[2]);
+	    sa_pos_data();
+	}
+	    
+	void ra_config_cmd(LCM_INPUT, const RAConfigCmd *msg){
+	    /*(*controllers)["RA_0"]->config(msg->Kp[0], msg->Ki[0], msg->Kd[0]);
+	    (*controllers)["RA_1"]->config(msg->Kp[1], msg->Ki[1], msg->Kd[1]);
+	    (*controllers)["RA_2"]->config(msg->Kp[2], msg->Ki[2], msg->Kd[2]);
+	    (*controllers)["RA_3"]->config(msg->Kp[3], msg->Ki[3], msg->Kd[3]);
+	    (*controllers)["RA_4"]->config(msg->Kp[4], msg->Ki[4], msg->Kd[4]);
+	    (*controllers)["RA_5"]->config(msg->Kp[5], msg->Ki[5], msg->Kd[5]);*/
+	}
+	    
+	void sa_config_cmd(LCM_INPUT, const SAConfigCmd *msg){
+	    (*controllers)["SA_0"]->config(msg->Kp[0], msg->Ki[0], msg->Kd[0]);
+	    (*controllers)["SA_1"]->config(msg->Kp[1], msg->Ki[1], msg->Kd[1]);
+	    (*controllers)["SA_2"]->config(msg->Kp[2], msg->Ki[2], msg->Kd[2]);
+	}
+	    
+	void hand_openloop_cmd(LCM_INPUT, const HandCmd *msg){
+	    (*controllers)["HAND_FINGER_POS"]->open_loop(msg->finger);
+	    (*controllers)["HAND_FINGER_NEG"]->open_loop(msg->finger);
+	    (*controllers)["HAND_GRIP_POS"]->open_loop(msg->grip);
+	    (*controllers)["HAND_GRIP_NEG"]->open_loop(msg->grip);
+	}
+
+	void gimbal_cmd(LCM_INPUT, const GimbalCmd *msg){
+	    (*controllers)["GIMBAL_PITCH_0_POS"]->open_loop(msg->pitch[0]);
+	    (*controllers)["GIMBAL_PITCH_0_NEG"]->open_loop(msg->pitch[0]);
+	    (*controllers)["GIMBAL_PITCH_1_POS"]->open_loop(msg->pitch[1]);
+	    (*controllers)["GIMBAL_PITCH_1_NEG"]->open_loop(msg->pitch[1]);
+	    (*controllers)["GIMBAL_YAW_0_POS"]->open_loop(msg->yaw[0]);
+	    (*controllers)["GIMBAL_YAW_0_NEG"]->open_loop(msg->yaw[0]);
+	    (*controllers)["GIMBAL_YAW_1_POS"]->open_loop(msg->yaw[1]);
+	    (*controllers)["GIMBAL_YAW_1_NEG"]->open_loop(msg->yaw[1]);
+	}
+	
+	void refreshAngles() {
+	    (*controllers)["RA_0"]->angle();
+	    (*controllers)["RA_1"]->angle();
+	    (*controllers)["RA_2"]->angle();
+	    (*controllers)["RA_3"]->angle();
+	    (*controllers)["RA_4"]->angle();
+	    (*controllers)["RA_5"]->angle();
+	    (*controllers)["SA_0"]->angle();
+	    (*controllers)["SA_1"]->angle();
+	    (*controllers)["SA_2"]->angle();
+	}
+
+	void ra_pos_data(){
+		ArmPosition msg;
+	    msg.joint_a = (*controllers)["RA_0"]->currentAngle;
+	    msg.joint_b = (*controllers)["RA_1"]->currentAngle;
+	    msg.joint_c = (*controllers)["RA_2"]->currentAngle;
+	    msg.joint_d = (*controllers)["RA_3"]->currentAngle;
+	    msg.joint_e = (*controllers)["RA_4"]->currentAngle;
+	    msg.joint_f = (*controllers)["RA_5"]->currentAngle;
+	    lcmBus->publish("/arm_position", &msg);
+
```

Comment:
NIT: Remove this blank line (i.e. no more than 1 blank line is needed here).

Reply:
implemented